diff -Naur a/src/3rdparty/webkit/Source/WebCore/platform/graphics/qt/FontPlatformDataQt.cpp b/src/3rdparty/webkit/Source/WebCore/platform/graphics/qt/FontPlatformDataQt.cpp
--- a/src/3rdparty/webkit/Source/WebCore/platform/graphics/qt/FontPlatformDataQt.cpp	2015-01-16 15:27:05.479152810 -0700
+++ b/src/3rdparty/webkit/Source/WebCore/platform/graphics/qt/FontPlatformDataQt.cpp	2015-01-16 15:30:09.048377591 -0700
@@ -73,7 +73,7 @@
     font.setLetterSpacing(QFont::AbsoluteSpacing, letterSpacing);
     const bool smallCaps = description.smallCaps();
     font.setCapitalization(smallCaps ? QFont::SmallCaps : QFont::MixedCase);
-    font.setStyleStrategy(QFont::ForceIntegerMetrics);
+    font.setStyleStrategy(QFont::PreferAntialias);

     m_data->bold = font.bold();
     // WebKit allows font size zero but QFont does not. We will return
