diff --git a/qt-4.8.5.old/src/gui/embedded/qscreen_qws.cpp b/qt-4.8.5.new/src/gui/embedded/qscreen_qws.cpp
index 0ae1667..cca52c6 100644
--- a/qt-4.8.5.old/src/gui/embedded/qscreen_qws.cpp
+++ b/qt-4.8.5.new/src/gui/embedded/qscreen_qws.cpp
@@ -1210,7 +1210,7 @@ void qt_blit_setup(QScreen *screen, const QImage &image,
         if (screen->pixelType() == QScreen::NormalPixel)
             screen->d_ptr->blit = blit_qrgb888;
         else
-            screen->d_ptr->blit = blit_24;
+           screen->d_ptr->blit = blit_template<qabgr888, quint24>; 
         break;
 #endif
 #ifdef QT_QWS_DEPTH_18
